hist/histv7/doc/DesignImplementation.md

```diff
@@ -0,0 +1,73 @@
+# Design and Implementation
```

Comment:
Maybe "Interface Design"? Unless that will get extended by more aspects.

Reply:
I think that's fine for now, we'll see later how the documentation bits develop.